src/Voron/Impl/Journal/WriteAheadJournal.cs

```diff
@@ -873,15 +878,23 @@ public void ApplyLogsToDataFile(CancellationToken token, TimeSpan timeToWait)
                             try
                             {
                                 _flushLockTaskResponsible.RunTaskIfNotAlreadyRan();
-                                if (_waitForJournalStateUpdateUnderTx.Wait(TimeSpan.FromMilliseconds(250), token))
-                                    break;
```

Comment:
Removing the `break` causes that we'll call `continue` below so we might not exit the loop (while currently we break it immediately). I'm not saying it's wrong. It's just a concern that requires double verification.

Reply:
After the `continue` we check the state here:
https://github.com/ravendb/ravendb/blob/52b5b145d0bd5dc39a5a629cc3e750febb4bf5f0/src/Voron/Impl/Journal/WriteAheadJournal.cs#L911

If the `_updateJournalStateAfterFlush` was updated we'll break (similar to the previous behavior).

Reply:
But I think we could do better. Instead of using single MRE (`_waitForJournalStateUpdateUnderTx`) for slightly different purposes we could have 2 dedicated MREs I think. 

```
var satisfiedIndex = WaitHandle.WaitAny(new[] { _waitForJournalStateUpdateUnderTx.WaitHandle, _onWriteTransactionCompleted.WaitHandle, token.WaitHandle }, TimeSpan.FromMilliseconds(250));

if (satisfiedIndex is 0 or 2) // _waitForJournalStateUpdateUnderTx or cancellation token
    break;
else if (satisfiedIndex is 1 or WaitHandle.WaitTimeout) // _onWriteTransactionCompleted or timeout
    continue;
else
    throw new InvalidOperationException($"Unknown handle at index: {satisfiedIndex}");
```

Reply:
done